packages/babel-helpers/src/helpers.js

```diff
@@ -413,6 +416,29 @@ helpers.interopRequireWildcard = template(`
   })
 `);
 
+helpers.specRequireInterop = template(`
+  (function (obj) {
+    if (obj && obj.__esModule) {
+      return obj;
+    } else {
+      var newObj = Object.create(null, {
+          default: {
+            value: obj,
+            writable: true,
+            enumerable: true
+          },
+          __esModule: {
```

Comment:
I would keep __esModule, you can check enumerability here to double check it is a "fake" es module. I am also concerned to keep w/ things that were already compiled in babel from the past.